### ISSUE TITLE
Add "Address" category for new node

### ISSUE THREAD

**Issue by rom1v** [organicmaps/organicmaps#9199]
**Is your feature request related to a problem? Please describe.**
I can assign an address number to a building, but I can't add an "Address" node (but I can see the existing ones).

**Describe the ideal solution**
When adding a place ("Add place to OpenStreetMap"), it asks for a category to select from a long list. "Address" should be one of the items of that list, to enter a street number.

**Describe alternatives you have considered**
- How do you solve this issue now with Organic Maps or other apps?
Either I need to assign the street number to a building, or I have to use another editor.

**Comment by biodranik** [comment]
Address addition as separate nodes is allowed only in some countries according to OSM rules, right? Which country are you in?

I think there were already related issues.

**Comment by rom1v** [comment]
> Address addition as separate nodes is allowed only in some countries according to OSM rules, right?

I don't know. I don't have much experience with OSM. But other streets in the same town have addresses as separate nodes.

> Which country are you in?

France.

**Comment by map-per** [comment]
> Address addition as separate nodes is allowed only in some countries according to OSM rules, right? Which country are you in?

No, there are no restrictions. It's more a question of taste whether you prefer addresses on buildings or on separate nodes. 
But there are regional differences, that are not really discoverable for users. E.g. in Germany POIs and addresses are always mapped separately, but in the Netherlands they are usually combined.

**Comment by muralito** [comment]
> > Address addition as separate nodes is allowed only in some countries according to OSM rules, right? Which country are you in?
> 
> No, there are no restrictions. It's more a question of taste whether you prefer addresses on buildings or on separate nodes. But there are regional differences, that are not really discoverable for users. E.g. in Germany POIs and addresses are always mapped separately, but in the Netherlands they are usually combined.

I always try to map address separately as a node, because is what is more resilient when users delete a POI, because the POI isn't anymore there, but the address information should remain.

**Comment by biodranik** [comment]
In many countries, addresses are bound to buildings, so adding them to buildings (or inside buildings) works well.

**Comment by map-per** [comment]
I think we should enable the Address category in`editor.config`. Our current approach for the editor is to enable everything unless there is a good reason for disabling it. For Address there is no good reason for hiding it.

Enabelling will also help with #6733 (People add to OSM the most "related" feature type / tag if the one they look for is missing)

Previous discussions about the Address category: #6394 #2371

**Comment by map-per** [comment]
I tried to implement this today, but it turned out to be more complicated than I expected. Addresses have the internal OM type `building|address` which causes the editor to add the nonsense OSM tag `building=address` to all address nodes.

Test Changeset: https://master.apis.dev.openstreetmap.org/node/4352691718

**Comment by map-per** [comment]
I think it would be OK to hard code an exception for address (like for recycling container and recycling ...) because addresses are also a special case in OSM. There is no tag to classify something as an address node. Addresses are defined by the absence of other `amenity=*`, `shop=*`, ... tags.

**Comment by road-runner-osm** [comment]
Beneath the Italy case (#4355), address points are helpful when the building is still missing. This is the case when the aerial image is not already updated and no building could be mapped. Afaik in most countries its okay to have only address points. If not, the address can be merged afterwards with the building.

**Comment by map-per** [comment]
I have a working version (https://github.com/map-per/organicmaps/tree/address2), but it still needs polishing before creating a PR. (The PR builds on top of #9242)
